libraries/AP_Mount/AP_Mount.cpp

```diff
@@ -422,6 +422,56 @@ void AP_Mount::handle_gimbal_manager_set_attitude(const mavlink_message_t &msg){
     }
 }
 
+void AP_Mount::handle_command_gimbal_manager_set_pitchyaw(const mavlink_message_t &msg){
+    mavlink_gimbal_manager_set_pitchyaw_t packet;
+    mavlink_msg_gimbal_manager_set_pitchyaw_decode(&msg,&packet);
+
+    AP_Mount_Backend *backend;
+
+    // check gimbal device id.  0 is primary, 1 is 1st gimbal, 2 is
+    // 2nd gimbal, etc
+    const uint8_t instance = packet.gimbal_device_id;
+    if (instance == 0) {
+        backend = get_primary();
+    } else {
+        backend = get_instance(instance - 1);
+    }
+
+    if (backend == nullptr) {
+        return;
+    }
+
+    // check flags for change to RETRACT
+    uint32_t flags = (uint32_t)packet.flags;
+    if ((flags & GIMBAL_MANAGER_FLAGS_RETRACT) > 0) {
+        backend->set_mode(MAV_MOUNT_MODE_RETRACT);
+        return;
+    }
+    // check flags for change to NEUTRAL
+    if ((flags & GIMBAL_MANAGER_FLAGS_NEUTRAL) > 0) {
+        backend->set_mode(MAV_MOUNT_MODE_NEUTRAL);
+        return;
+    }
+
+    // param5 : pitch_angle (in radians)
+    // param6 : yaw angle (in radians)
+    const float pitch_angle_deg = RadiansToCentiDegrees(packet.pitch)/100.0f;
+    const float yaw_angle_deg = RadiansToCentiDegrees(packet.yaw)/100.0f;
+    if (!isnan(pitch_angle_deg) && !isnan(yaw_angle_deg)) {
+        set_angle_target(instance, 0, pitch_angle_deg, yaw_angle_deg, flags & GIMBAL_MANAGER_FLAGS_YAW_LOCK);
+        return;
+    }
+
+    // param7 : pitch_rate (in rad/s)
+    // param8 : MAV_RESULT AP_Mount::yaw rate (in rad/s)
+    const float pitch_rate_degs = RadiansToCentiDegrees(packet.pitch_rate)/100.0f;
+    const float yaw_rate_degs = RadiansToCentiDegrees(packet.yaw_rate)/100.0f;
+    if (!isnan(pitch_rate_degs) && !isnan(yaw_rate_degs)) {
```

Comment:
@peterbarker,

Is it ok to do a mathematical operation on a float that is potentially NaN and then check if it is a NaN?  I think this will work but if it was me I would probably do the NaN check first.  What do you think?

Reply:
I agree - do the NaN check before the maths.

... also, should be using `degrees(pitch_rate)`, not `RadiansToCentiDegrees(pitch_rate)/100.0f`